lib/type/any.ex

```diff
@@ -1,20 +1,34 @@
 defmodule Litmus.Type.Any do
-  @moduledoc """
-  Schema and validation for Any data type.
-  """
+  @moduledoc false
 
   defstruct required: false
 
   @type t :: %__MODULE__{
           required: boolean
         }
-end
 
-defimpl Litmus.Type, for: Litmus.Type.Any do
-  alias Litmus.Type
+  @spec add_required_errors(map, binary, t) :: {:ok, map} | {:error, String.t()}
+  def add_required_errors(params, field, %Litmus.Type.Any{required: true}) do
+    if Map.has_key?(params, field) do
+      {:ok, params}
+    else
+      {:error, "#{field} is required"}
+    end
+  end
+
+  def add_required_errors(params, _field, %Litmus.Type.Any{required: false}), do: {:ok, params}
+
+  def add_required_errors(_params, _field, %Litmus.Type.Any{required: _}) do
+    {:error, "Any.required must be a boolean"}
+  end
+
+  @spec validate_keys(map, binary, t) :: {:ok, map} | {:error, String.t()}
+  def validate_keys(data, field, type), do: add_required_errors(data, field, type)
```

Comment:
I'd move this above `add_required_errors` because it is the top level function of this types' validation